runner/public/static/national-insurance.js

```diff
@@ -0,0 +1,3 @@
+$("input[type='ni']").on('input propertychange paste', function () {
```

Comment:
interesting, how does this work and what exactly does it do?

Reply:
a wild jQueryism appeared! (forgot how ugly/unhelpful it can be until I saw this). onpaste, it's a html event property, like onclick. changes made are being listened to/bound to with jquery. I believe this might be the only way to read clipboard data cross browser. I wouldn't rely on this, but it's nice to have on the FE.

Reply:
The bit that makes me feel a bit wrong is that I've had to add a single js file and modify the layout to reference it. It's not minified and not bundled. There's a couple of other static js files that are following the same pattern to do some other JQuery magic. (not mine!!)

I'll add the server side behaviour to make sure this is backed up by some code that will always execute.